src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
To put it another way, the IEnumerable loop and the count++ is one concern and the itemConstraint filter is another concern. If we have a way to solve it so that those two concerns are separated into two classes that can compose, perfect! I hope that's the eventual solution. However, a short term solution of separating into two classes is more harmful than anything if both classes still retain the IEnumerable loop counting concern. In that case not only does the class with the filter still deal with both concerns, but now you have duplicated code which results in invisible behavioral coupling between two classes instead of coupling between the branches of the itemConstraint null check.